### PR TITLE
paypal charge

### PR DESCRIPTION
**Pull request by aykamko**
#165

### CODE REVIEW

File: revolv/payments/lib/instruments.py [hunk 1]
```diff
@@ -59,12 +66,14 @@ def __init__(self, credit_card):
 
     def charge(self, amount):
         """
-        Charge the credit card for the given amount.
+        Documentation: https://github.com/paypal/PayPal-Python-SDK
+
+        Charge the credit card for the given amount, floored to 2 decimal
+        places. Raises InvalidCreditCardException if charge fails.
 
         :param amount:
```
Comment (noahsark769) [inline revolv/payments/lib/instruments.py]: Say that this needs to be a float :)

Comment (noahsark769) [inline revolv/payments/lib/instruments.py]: Or at least add a test that makes sure it works with both ints and floats

Comment (aykamko) [inline revolv/payments/lib/instruments.py]: Note to self: add tests for int, float, and Decimal

**Comment by aykamko** [comment]
Note to self: add gmail and Paypal Dev credentials to documentation before closing pull

File: revolv/project/tests/tests_general.py [hunk 2]
```diff
@@ -298,6 +298,7 @@ def perform_valid_donation(self):
             HTTP_X_REQUESTED_WITH='XMLHttpRequest'
         )
 
+    @override_settings(ENABLE_PAYMENT_CHARGING=False)
```
Comment (noahsark769) [inline revolv/project/tests/tests_general.py]: We should have a comment saying why this is necessary to override (i.e. that dummy card numbers will cause the paypal sdk stuff to fail)

File: revolv/templates/project/partials/payment_modal.js [hunk 3]
```diff
@@ -378,6 +389,13 @@ $('#donate-form').submit(function(e) {
         $donateModalErrors.children('.error-msg')
             .text('Your card information is invalid. Please correct it.');
         $('#donate-modal').foundation('reveal', 'open');
+    }).always(function () {
+        paymentSpinner.stop();
+        setTimeout(function() {
```
Comment (noahsark769) [inline revolv/templates/project/partials/payment_modal.js]: This deserves a comment saying why we need to do setTimeout (i.e. that if we don't the modal fading animation overlaps)

File: revolv/settings.py [hunk 4]
```diff
@@ -295,4 +291,5 @@
     ('en-us', 'English'),
 ]
 
-CHARGE_INSTRUMENT = False
+# be careful!
```
Comment (noahsark769) [inline revolv/settings.py]: I think we should expand this comment to talk about how the paypal keys work and how cards won't actually get charged unless there are non-sandbox keys. Also should put that somewhere more visible like the readme AND the docstring of PaymentInstrument

**Comment by noahsark769** [comment]
@aykamko seems like this PR should be good to go now, can you merge in master though?

**Comment by noahsark769** [comment]
NOICE

**Pull request merged**